### ISSUE TITLE
Unable to retrieve 'AutomatedTestId' (GUID) from WebApi.TestCaseResult

### ISSUE THREAD

**Issue by GraueEminenz79** [Microsoft/azure-pipelines-tasks#9418]
## Environment & Setup
- TFS on-premises TFS 2018 Update 3 (16.131.28226.3)
- Build Definition with 
  - Test execution creating **TRX** result file
  - _Publish Test Result_ task publishing it (result format: VSTest)
- Post-proccessing application querying TestRuns for test results of type
 `Microsoft.TeamFoundation.TestManagement.WebApi.TestCaseResult` with `TestManagementHttpClient.GetTestResultsAsync("someTeamProject", sometestRunId, ResultDetails.SubResults)`
  - nuget packages **16.144.1-preview**

## Issue Description
Until yesterday (on TFS 2017) I was able to read-out the following properties from `TestCaseResult`
```
AutomatedTestStorage
AutomatedTestType
AutomatedTestId
```
---  all filled properly.

Since the upgrade to TFS 2018 Update 3, `AutomatedTestId` (GUID) is **empty**.
Although the input TRX result file is unchanged and none of the nuget package version were changed.

> NOTE: All `AutomatedTestId` s generated and published to TFS BEFORE the upgrade are still available!

**Comment by GraueEminenz79** [comment]
tldr;
Column `AutomatedTestId` of Table `[TestResult].[tbl_TestCaseReference]` is not getting filled anymore during publish.

**Comment by nigurr** [comment]
@GraueEminenz79 We don't support RC builds as RTM builds are already out.
Please let us know if you are still facing issue with TFS 2018 Update 3.

**Comment by GraueEminenz79** [comment]
@nigurr My bad, speaking of RC, version **is** actually 16.131.28226.3 = TFS 2018 Update 3, right!?

**Comment by nigurr** [comment]
@GraueEminenz79 Thanks for info. Seems like a regression, will be assigning to right team to help you with this.

**Comment by GraueEminenz79** [comment]
I downloaded nuget package https://www.nuget.org/packages/Microsoft.TeamFoundation.PublishTestResults/16.144.1-preview
to have a look into the `TrxResultParser` to find out that `TestCaseResultData` looked fine.
Parsing of TRX seems to be off the table.
I must admit, I know nothing of the publishing to TFS datatables itself....

**Comment by SadagopanRajaram** [comment]
Based on usage, this was a conscious decision in TFS 2018. Automated Test ID per our understanding was used primarily for association with the planned test case and was guaranteed to be unique. However, in other runners, the execution ID was sometimes duplicated for the same test method which potentially could break uniqueness assumptions for reporting scenarios. Since there wasn't any known scenario, we took a call to support the automated test ID only when results were associated with a test plan. If your scenario is similar but you need to associate NUnit tests with a test plan, we just added support for this and you can use that instead.  Please go through the FAQ in the documentation below
https://docs.microsoft.com/en-us/azure/devops/test/associate-automated-test-with-test-case?view=azure-devops. 
Please reactivate if this does not address your complete scenario and we can work on resolving it.
